sdk/textanalytics/ai-text-analytics/test/textAnalyticsClient.spec.ts

```diff
@@ -105,11 +108,112 @@ describe("[AAD] TextAnalyticsClient", function() {
+          assert.equal("premium", premiumOpinion.text);
+          assert.equal("positive", premiumOpinion.sentiment);
+          assert.isAtLeast(premiumOpinion.confidenceScores.positive, 0);
+          assert.isAtLeast(premiumOpinion.confidenceScores.neutral, 0);
+          assert.isAtLeast(premiumOpinion.confidenceScores.positive, 0);
+          assert.isFalse(premiumOpinion.isNegated);
+        })
+      );
+    });
+
+    it("client gets negative mined opinions", async () => {
+      const documents = [
+        {
+          text: "The food and service is not good",
+          id: "0",
+          language: "en"
+        }
+      ];
+      const results: AnalyzeSentimentResultArray = await client.analyzeSentiment(documents, {
+        mineOpinions: true
+      });
+      assert.equal(results.length, 1);
+      assertAllSuccess(results);
+      const documentSentiment: AnalyzeSentimentSuccessResult = results[0] as AnalyzeSentimentSuccessResult;
+      documentSentiment.sentences.map((sentence) => {
+        const foodAspect = sentence.minedOpinions?.[0].aspect;
+        assert.equal("food", foodAspect?.text);
+        assert.equal("negative", foodAspect?.sentiment);
+        assert.isAtLeast(foodAspect?.confidenceScores.positive!, 0);
+        assert.isUndefined(foodAspect?.confidenceScores.neutral);
+        assert.isAtLeast(foodAspect?.confidenceScores.negative!, 0);
+
+        const serviceAspect = sentence.minedOpinions?.[0].aspect;
+        assert.equal("food", serviceAspect?.text);
+        assert.equal("negative", serviceAspect?.sentiment);
+        assert.isAtLeast(serviceAspect?.confidenceScores.positive!, 0);
+        assert.isUndefined(serviceAspect?.confidenceScores.neutral);
+        assert.isAtLeast(serviceAspect?.confidenceScores.negative!, 0);
+
+        const foodOpinion = sentence.minedOpinions?.[0].opinions[0];
+        const serviceOpinion = sentence.minedOpinions?.[1].opinions[0];
+
+        assertOpinionsEqual(foodOpinion!, serviceOpinion!);
+        
+        assert.equal("good", foodOpinion?.text);
+        assert.equal("negative", foodOpinion?.sentiment);
+        assert.isAtLeast(foodOpinion?.confidenceScores.positive!, 0);
+        assert.isAtLeast(foodOpinion?.confidenceScores.neutral!, 0);
+        assert.isAtLeast(foodOpinion?.confidenceScores.positive!, 0);
+        assert.isTrue(foodOpinion?.isNegated);
+      });
+    });
+
+    it("client gets no mined opinions", async () => {
+      const documents = [
+        {
+          text: "today is a hot day",
+          id: "0",
+          language: "en"
+        }
+      ];
+      const results: AnalyzeSentimentResultArray = await client.analyzeSentiment(documents, {
+        mineOpinions: true
+      });
+      assert.equal(results.length, 1);
+      assertAllSuccess(results);
+      const documentSentiment: AnalyzeSentimentSuccessResult = results[0] as AnalyzeSentimentSuccessResult;
+      assert.isEmpty(documentSentiment.sentences[0].minedOpinions);
```

Comment:
I might be wrong here (not very knowledgeable in JS), but the type is nullable `minedOpinions?`, so shouldn't this be null instead of empty?

Reply:
`minedOpinions` is a list and in this case it is empty.

Reply:
ohh I see. So when `includeOpinionMining` is sent and no results are returned we have an empty list.
If the flag is not set then the object itself is null, right?
If so, could we add a test to verify that last behavior?

Reply:
done.